File: app/src/main/java/control/Screen.java
package control;

import javax.swing.*;

public class Screen extends JFrame {

    Screen(RuntimeSettings runtimeSettings) {
        super(runtimeSettings.getWindowTitle());
        this.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        this.setSize(runtimeSettings.getScreenWidth(), runtimeSettings.getScreenHeight());
        this.setResizable(isResizable());
    }
}

File: app/src/main/kotlin/graphics/Screen.kt
package graphics

import control.RuntimeSettings
import javax.swing.JFrame

class Screen internal constructor(
    private val runtimeSettings: RuntimeSettings
) : JFrame(runtimeSettings.windowTitle) {

    fun init() {
        this.defaultCloseOperation = EXIT_ON_CLOSE
        this.setSize(runtimeSettings.screenWidth, runtimeSettings.screenHeight)
        this.isResizable = runtimeSettings.isResizable
        this.isVisible = true
    }
}
